src/syntax/tokenizer.rs

```diff
@@ -140,7 +140,8 @@ impl Tokenizer {
         Self {
             pos: Cell::new(Position::new(0, 1, 1)),
             // Starts as `true` because the first token must always be a contract name.
-            identifier_mode: Cell::new(true),
+            // identifier_mode: Cell::new(true), // @follow-up - this is correct, but it breaks the tests because of invalid . character in root
```

Comment:
The `.`s in a root should be valid now, since I am calling `sanitize` on contract names. We have multiple tests for this, but the main one that covers this functionality you can find here:

https://github.com/alexfertel/bulloak/blob/c38e7795c6a018019ed2e2e0712868b5db0202be/tests/scaffold.rs#L12

Which, scaffolds this file:

https://github.com/alexfertel/bulloak/blob/c38e7795c6a018019ed2e2e0712868b5db0202be/tests/scaffold/basic.tree

Which contains a contract name with a `.`.

Reply:
@alexfertel, based on the latest commit, all tests pass. The issue above occurs when changing the `reset` function to set `identifier_mode` to true. The rationale here that, given `identifier_mode` is set to true by default since:
> The first token must always be a contract name, which has to be a valid Solidity identifier.

it should also be true here as we don't want this state to be changed to false when `reset` is called within `tokenize`; however, since the contract name is not sanitized before the tokenization step then the tests fail with `IdentifierCharInvalid('.')`.

Do you agree with this assessment? If so, how should we handle this?

Reply:
Yes, you were right; it's fine not to treat contract names as identifiers since we sanitize them later. Lmk if you find any bugs with this behavior.